### DISCUSSION TITLE
How to check for and install .Net 7

### DISCUSSION THREAD

**Discussion by mjanulaitis** [wixtoolset/issues#7128] [Questions] [answered]
I am creating my first Wix4 project for a new .Net 7 application that runs on Windows.  Does anyone know how to use Wix 4 to check for .Net 7 then, if missing, install from the web and if air gapped throw an error message?

**Comment by GF-Huang** [comment]
I don't even know where the documentation for WIX v4 is, do you know?

**Comment by BMurri** [comment, marked as answer]
Start [here](https://wixtoolset.org/docs/reference/wixext/dotnet/#net-and-net-core)?

**Reply by robmen** [reply]
To be fair, some of that documentation may have been written after Jan 30.

**Reply by BMurri** [reply]
That may have played a part in my not answering this question then. But, these questions have multiple audiences.

**Reply by GF-Huang** [reply]
Maybe I misspoken, documentation is not equal to tutorial. Actually what I want to ask is where is the tutorial of v4.

**Comment by BMurri** [comment]
If memory serves, tutorial for v3 took what, 12-18 months after release, by someone who wasn't part of the core team? Then again, that was over a decade ago.

Given core team constraints, the focus right now is on getting v4 out the door. The biggest fans then tend to push in some documentation and put the meat on the tutorials/sample projects/etc. I'm pretty sure it won't take as long this time for the tutorial to show up.

In the meantime, the Development Dojo is doing an incredible job. It's on YouTube. Check it out!

**Reply by robmen** [reply]
Blair is on the right track. My company, [FireGiant](https://www.firegiant.com/), is responsible for [the tutorial](https://www.firegiant.com/wix/tutorial/) as the original author stepped away many years ago. Updating the tutorial to WiX v4 is very high on our backlog, but getting WiX v4 and [HeatWave](https://www.firegiant.com/wix/heatwave/) released remains higher.

The [Deployment Dojo](https://www.youtube.com/playlist?list=PLDlzbQXIs18slmqmdlS10_de_Cps-QRg6) is something I do for fun and I'm happy if it provides useful information.

**Reply by GF-Huang** [reply]
OH, Bruce Lee fans.

**Reply by robmen** [reply]
🏯 He provided a great quote that speaks to the power _and_ flexibility of the WiX Toolset and the Windows Installer.
